### PR TITLE
Notebook invoker APIM API

### PR DESCRIPTION
**Pull request by masoncusack**
Given a Terraform APIM resource as input, an API is automatically deployed to invoke the Jupyter Notebook as a [Databricks job](https://docs.databricks.com/jobs.html).

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -86,3 +91,299 @@ resource "databricks_notebook" "notebook" {
   overwrite = false
   format    = "JUPYTER"
 }
+
+resource "azurerm_api_management_api" "notebook_api" {
```
Comment (jjcollinge) [inline main.tf:95]: As the notebook is conditional - does this API not need to be too?

Comment (masoncusack) [reply main.tf:95]: Good spot. Made API conditional on notebook_path variable and tested manually. 🙂

File: main.tf [hunk 2]
```diff
@@ -86,3 +91,300 @@ resource "databricks_notebook" "notebook" {
...
+  name                = "invoke_notebook"
+  resource_group_name = var.apim.resource_group_name
+  api_management_name = var.apim.name
+  revision            = "1"
+  display_name        = "Notebook invocation API"
+  path                = "run"
+  protocols           = ["https"]
+  service_url         = "${local.db_host}/api/2.0/jobs"
+  count     = var.notebook_path == "" ? 0 : 1
```
Comment (stevehaigh) [inline main.tf:104]: nit pic - formatting :)

**Review by stevehaigh — APPROVED** [review]

File: main.tf (hunk 2 above)
Comment (masoncusack) [reply main.tf:104]: Will resolve with next PR.

**Pull request merged**